protostar/starknet_gateway/gateway_facade.py

```diff
@@ -281,3 +284,44 @@ def __init__(self, function_name: str):
 class ContractNotFoundException(ProtostarException):
     def __init__(self, contract_address: AddressRepresentation):
         super().__init__(f"Tried to call unknown contract:\n{contract_address}")
+
+
+class InvalidInputException(ProtostarException):
+    pass
+
+
+def validate_deploy_input(compiled_contract_json: str, inputs: List[int]) -> None:
+    def constructor_filter(element: Dict):
+        return element["type"] == "constructor"
+
+    def get_type_size(abi: List[Dict], typename: str):
+        if typename == "felt":
+            return 1
+
+        def type_filter(element: Dict):
+            return element["name"] == typename
+
+        [type_definition] = filter(type_filter, abi)
+        return type_definition["size"]
+
+    abi = json.loads(compiled_contract_json)["abi"]
+    [constructor] = filter(constructor_filter, abi)
```

Comment:
1. List comprehension would be more pythonic here.
2. Is there a possibility that there are two constructors in ABI?

Reply:
1. done